### ISSUE TITLE
Imports are skipping units used only by <cn> terms

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#519]
When you import a component containing units which are *only* referenced by the <cn> terms in the maths, these units are not being imported and the validator returns a "units not found" error. 
 I'll make a PR with the failing test.

eg:  
```
TEST(Model, missingUnitsFromImportOfCnTerms)
{
    std::string a =
        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"myModel\">\n"
        "   <units name=\"myUnitsThatReallyExist\">\n"
        "       <unit exponent=\"-1\" units=\"second\"/>\n"
        "   </units>\n"
        "   <component name=\"myComponent\">\n"
        "       <variable name=\"a\" units=\"second\"/>\n"
        "       <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
        "           <apply><eq/>\n"
        "               <ci>a</ci>\n"
        "               <cn cellml:units=\"myUnitsThatReallyExist\">1</cn>\n"
        "           </apply>\n"
        "       </math>\n"
        "   </component>\n"
        "</model>";
    auto parser = libcellml::Parser::create();
    auto modelFromString = parser->parseModel(a);
    auto validator = libcellml::Validator::create();
    validator->validateModel(modelFromString);
    EXPECT_EQ(size_t(0), validator->errorCount());

    // Import the membrane component from a file
    auto model = libcellml::Model::create("model_from_imports");
    auto c = libcellml::Component::create("c");

    auto imp = libcellml::ImportSource::create();
    imp->setUrl("units_in_cn.cellml");

    c->setImportReference("myComponent");
    c->setImportSource(imp);
    model->addComponent(c);


    EXPECT_TRUE(model->hasUnresolvedImports());
    model->resolveImports(resourcePath());
    EXPECT_FALSE(model->hasUnresolvedImports());
    model->flatten();

    validator->validateModel(model);
    for(size_t i = 0; i < validator->errorCount(); ++i){
        std::cout << validator->error(i)->description()<<std::endl;
    }
    EXPECT_EQ(size_t(0), validator->errorCount());
}
```

**Comment by hsorby** [comment]
If you find yourself printing out errors from a `Validator` or related class, there is a test utility `printErrors` that will do this job.  Saves a bit of typing.

**Comment by kerimoyle** [comment]
Yes, thanks @hsorby .  I used it originally, then thought it wasn't working as I was getting no output ... turned out it was exiting earlier.  But yep, will use it next time.  Any thoughts on the behaviour?  Are we expecting people to import units explicitly here, or that they'd be pulled in with the overall import of the component?

**Comment by hsorby** [comment]
Any units an imported component requires should also be brought in whether the units themselves were imported or not I thought I had a test for this but apparently not.

**Comment by hsorby** [comment]
No I don't have a test for that scenario having looked properly at it, should be an easy fix though.

**Comment by kerimoyle** [comment]
Fixed in #520
